VoxelSniperCore/src/main/java/com/github/kevindagame/util/BlockWrapper.java

```diff
@@ -100,7 +134,23 @@ public final int getZ() {
     /**
      * @param z the z to set
      */
-    public final void setZ(final int z) {
+    public final BlockWrapper setZ(final int z) {
         this.z = z;
+        return this;
+    }
+
+    @Override
+    public BlockWrapper clone() {
+        try {
+            BlockWrapper clone = (BlockWrapper) super.clone();
+            clone.blockData = this.blockData.getMaterial().createBlockData();
+            clone.x = this.x;
+            clone.y = this.y;
+            clone.z = this.z;
+            clone.world = this.world;
```

Comment:
```suggestion
            clone.blockData = this.blockData.getMaterial().createBlockData();
```
Don't think we need to re-assign x, y, z and world.
Should we make IBlockData Cloneable as well? The way you do it right now will lose the data.

Reply:
looking at this, we are losing any data that is contained in BlockData (for example the age of wheat).